### PR TITLE
chore(wallet): implement DApps API Tx actions

### PR DESCRIPTION
**Pull request by Pavneet-Sing**
- integrate and show DApps Tx requests
- implement reject all
- implement approval of all pending tx

Resolves https://github.com/brave/brave-browser/issues/22460

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Note: For demo, the dialog is being triggered after wallet is closed, after unlock.

Reject all Tx: 

https://user-images.githubusercontent.com/11755381/164461948-49af5b9f-6730-4dee-972d-96d91d68fdff.mp4

Approve Tx:

https://user-images.githubusercontent.com/11755381/164462807-6c4b4c9d-b64a-4ce9-91e3-e039a908955c.mp4

### CODE REVIEW

File: browser/ui/android/strings/android_brave_strings.grd [hunk 1]
```diff
@@ -2953,6 +2953,8 @@ If you don't accept this request, VPN will not reconnect and your internet conne
     <message name="IDS_PERMISSION_WAS_CANCELLED" desc="Vpn user consent text">
       Permission was cancelled
     </message>
+    <message name="IDS_BRAVE_WALLET_QUEUE_REJECT_ALL" desc="Confirm Transaction Queue Reject All Transactions">Reject <ph name="TRANSACTION_LENGTH"><ex>4</ex>%1$s</ph> transactions</message>
+
```
Comment (deeppandya) [inline browser/ui/android/strings/android_brave_strings.grd:2957]: nit : I don't think we require new line here.

**Review by deeppandya — APPROVED** [review]
nice way to expose livedata. LGTM

File: browser/ui/android/strings/android_brave_strings.grd (hunk 1 above)
Comment (SergeyZhukovsky) [reply browser/ui/android/strings/android_brave_strings.grd:2957]: could you please format it also as all strings formatted?

Comment (Pavneet-Sing) [reply browser/ui/android/strings/android_brave_strings.grd]: done

**Review by qamarngr — APPROVED** [review]

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Pull request merged**